anndata/_io/specs/methods.py

```diff
@@ -481,6 +482,35 @@ def read_sparse_partial(elem, *, items=None, indices=(slice(None), slice(None)))
     return SparseDataset(elem)[indices]
 
 
+#################
+# Awkward array #
+#################
+
+
+@_REGISTRY.register_write(H5Group, AwkArray, IOSpec("awkward-array", "0.1.0"))
+@_REGISTRY.register_write(ZarrGroup, AwkArray, IOSpec("awkward-array", "0.1.0"))
+def write_awkward(f, k, v, dataset_kwargs=MappingProxyType({})):
+    import awkward._v2 as ak
+
+    group = f.create_group(k)
+    form, length, container = ak.to_buffers(v)
+    group.attrs["length"] = length
+    group.attrs["form"] = form.tojson()
+    write_elem(group, "container", container, dataset_kwargs=dataset_kwargs)
```

Comment:
v2 api updates + packing the array

```suggestion
    form, length, container = ak.to_buffers(ak.packed(v))
    group.attrs["length"] = length
    group.attrs["form"] = form.to_json()
    write_elem(group, "container", container, dataset_kwargs=dataset_kwargs)
```